### DISCUSSION TITLE
test with vitest

### DISCUSSION THREAD

**Discussion by datahoecn** [prisma/prisma#13580] [Q&A] [answered]
Vitest is a very popular testing framework, I want to use vitest instead of jest for prisma mock testing!

this is prisma unit test page: https://www.prisma.io/docs/guides/testing/unit-testing

test uses jest-mock-extended, in vitest replace with vitest-mock-extended

here my test code, mock not work!

```
import { PrismaClient } from '@prisma/client';
import { mockDeep, mockReset } from 'vitest-mock-extended';
import { type DeepMockProxy } from 'vitest-mock-extended/lib/Mock';

import prisma from './client';

vi.mock('./client', () => ({
  __esModule: true,
  default: mockDeep<PrismaClient>(),
}));

// beforeEach(() => {
//   mockReset(prismaMock);
// });

export const prismaMock = prisma as unknown as DeepMockProxy<PrismaClient>;

it('test prisma', async () => {
  const user = {
    id: 1,
    user_id: 1,
    created_time: new Date(),
    updated_time: new Date(),
  };
  prismaMock.payingUser.create.mockResolvedValue(user);
  const user1 = await prismaMock.payingUser.create.(user); // can work, no crud in db
  const user2 = await prisma.payingUser.create({data: user}); // not work, crud in db
  console.log(user1, user2);
});
```

**Comment by datahoecn** [comment, marked as answer]
I use vitest env solve this problem, using prismaMock in the test env and prisma in the dev env, just like this:

```
if (import.meta.env.PROD) PrismaClient = ProdPrisma.PrismaClient;
let prisma = new PrismaClient({
  log: ['query', 'info', 'warn', 'error'],
});
if (import.meta.env.DEV) prisma = prisma;
if (import.meta.env.test) prisma = prismaMock;
export { prisma };
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
